### PR TITLE
fix(lint/useJsxKeyInIterable): fix a false positive when key is in the return statement, but not any variable declarations

### LINKED ISSUE #3020: 💅 `useJsxKeyInIterable` shows warning even key is specified
### Environment information

```bash
CLI:
  Version:                      1.7.4-nightly.125f34b
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.0.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "bun/1.1.8"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0

 aristotle-ui  
 aristotle-ui  biome rage --linter
zsh: command not found: biome
 aristotle-ui  bunxbiome rage --linter
zsh: command not found: bunxbiome
 aristotle-ui  bunx biome rage --linter
CLI:
  Version:                      1.7.4-nightly.125f34b
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.0.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "bun/1.1.8"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  Recommended:                  true
  All:                          false
  Rules:                        a11y/useKeyWithClickEvents = "off"
                                complexity/noEmptyTypeParameters = "error"
                                complexity/noExcessiveCognitiveComplexity = {"level":"error","options":{"maxAllowedComplexity":30}}
                                complexity/noForEach = "off"
                                complexity/noStaticOnlyClass = "off"
                                complexity/noUselessFragments = "error"
                                complexity/noUselessLoneBlockStatements = "error"
                                complexity/useSimplifiedLogicExpression = "error"
                                correctness/noInvalidUseBeforeDeclaration = "error"
                                correctness/noUndeclaredVariables = "error"
                                correctness/noUnusedImports = "error"
                                correctness/noUnusedPrivateClassMembers = "error"
                                correctness/noUnusedVariables = "error"
                                correctness/useHookAtTopLevel = "error"
                                nursery/noDuplicateJsonKeys = "error"
                                nursery/noFlatMapIdentity = "error"
                                nursery/noNodejsModules = "error"
                                nursery/useSortedClasses = {"level":"error","options":{"attributes":["class","className"],"functions":["joinClasses"]}}
                                performance/noBarrelFile = "error"
                                style/noNamespaceImport = "error"
                                style/useCollapsedElseIf = "error"
                                style/useConsistentArrayType = {"level":"error","options":{"syntax":"generic"}}
                                style/useExportType = "error"
                                style/useFilenamingConvention = {"level":"error","options":{"filenameCases":["camelCase","export","PascalCase"]}}
                                style/useForOf = "error"
                                style/useFragmentSyntax = "error"
                                style/useImportType = "error"
                                style/useNamingConvention = "off"
                                style/useNodeAssertStrict = "error"
                                style/useNodejsImportProtocol = "error"
                                style/useNumberNamespace = "error"
                                style/useShorthandAssign = "error"
                                style/useShorthandFunctionType = "error"
                                style/useSingleCaseStatement = "error"
                                suspicious/noConsoleLog = "warn"
                                suspicious/noEmptyBlockStatements = "error"
                                suspicious/noGlobalAssign = "error"
                                suspicious/noThenProperty = "error"
                                suspicious/useAwait = "error"

Workspace:
  Open Documents:               0
```

### Rule name

useJsxKeyInIterable

### Playground link

https://biomejs.dev/playground/?lineWidth=120&code=aQBtAHAAbwByAHQAIAB7ACAARgByAGEAZwBtAGUAbgB0ACAAfQAgAGYAcgBvAG0AIAAnAHIAZQBhAGMAdAAnADsACgAKAGMAbwBuAHMAdAAgAGkAdABlAG0AcwAgAD0AIABbAF0AOwAKAAoAYwBvAG4AcwB0ACAAQwB1AHMAdABvAG0AUABvAHAAbwB2AGUAcgBNAGUAbgB1ACAAPQAgACgAKQAgAD0APgAgAHsACgAJAHIAZQB0AHUAcgBuACAAKAAKAAkACQA8AGQAaQB2AD4ACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAB7AGkAdABlAG0AcwAuAG0AYQBwACgAKABpAHQAZQBtACkAIAA9AD4AIAB7AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIABsAGUAdAAgAG4AbwBkAGUAIAA9ACAAKAA8AGIAdQB0AHQAbwBuACAAdAB5AHAAZQA9ACIAYgB1AHQAdABvAG4AIgA%2BAHsAaQB0AGUAbQAuAGwAYQBiAGUAbAB9ADwALwBiAHUAdAB0AG8AbgA%2BACkAOwAKAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIABpAGYAIAAoAGkAdABlAG0ALgBpAHMATABpAG4AawApACAAbgBvAGQAZQAgAD0AIAAoADwAYQAgAGgAcgBlAGYAPQAiAC8AIgA%2BAHsAbgBvAGQAZQB9ADwALwBhAD4AKQA7AAoACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAHIAZQB0AHUAcgBuACAAPABGAHIAYQBnAG0AZQBuAHQAIABrAGUAeQA9AHsAaQB0AGUAbQAuAGwAYQBiAGUAbAB9AD4AewBuAG8AZABlAH0APAAvAEYAcgBhAGcAbQBlAG4AdAA%2BADsACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAB9ACkAfQAKACAAIAAgACAAIAAgACAAIAA8AC8AZABpAHYAPgAKAAkAKQA7AAoAfQA7AAoA

### Expected result

It should not show warning/error when key is specified.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

### PR DESCRIPTION
**Pull request by dyc3**
## Summary

This PR fixes a false positive where the rule would flag variable declaration statements even if the return statement had a component with the `key` prop set.

fixes #3020

## Test Plan

Added snapshot test

### CODE REVIEW

**Comment by ematipico** [comment]
Are the names of your branches automatically generated? They are very long! 😆

File: crates/biome_js_analyze/src/lint/correctness/use_jsx_key_in_iterable.rs [hunk 1]
```diff
@@ -184,6 +184,28 @@ fn handle_function_body(
...
+        .find_map(|statement| statement.as_js_return_statement().cloned());
+    let is_return_component = return_statement
+        .as_ref()
+        .and_then(|ret| {
+            let returned_value = ret.argument()?;
+            Some(ReactComponentExpression::can_cast(
+                returned_value.syntax().kind(),
+            ))
+        })
+        .unwrap_or(false);
```
Comment (ematipico) [inline crates/biome_js_analyze/src/lint/correctness/use_jsx_key_in_iterable.rs:200]:
```suggestion
        .unwrap_or_default();
```

**Review by ematipico — APPROVED** [review]

**Review by anonrig — APPROVED** [review]
looks good after applying @ematipico's recommendation

**Comment by dyc3** [comment]
> Are the names of your branches automatically generated? They are very long! 😆

Yeah. I use graphite on my end to stack branches. It helps keep my PRs smaller (in general) to make it easier to review them, without necessarily having to wait for reviews to continue working on stuff.

**Pull request merged**